src/apify_client/clients/resource_clients/task.py

```diff
@@ -0,0 +1,214 @@
+from typing import Any, Dict, List, Optional, cast
+
+from ..._consts import ActorJobStatus
+from ..._errors import ApifyApiError, ApifyClientError
+from ..._utils import _catch_not_found_or_throw, _encode_json_to_base64, _filter_out_none_values_recursively, _parse_date_fields, _pluck_data
+from ..base.resource_client import ResourceClient
+from .webhook_collection import WebhookCollectionClient
+
+
+class TaskClient(ResourceClient):
+    """Sub-client for manipulating a single task."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        """Initialize the TaskClient."""
+        super().__init__(*args, resource_path='actor-tasks', **kwargs)
+
+    def get(self) -> Optional[Dict]:
+        """Retrieve the task.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-object/get-task
+
+        Returns:
+            The retrieved task
+        """
+        return self._get()
+
+    def update(
+        self,
+        *,
+        name: Optional[str] = None,
+        task_input: Optional[Dict] = None,
+        build: Optional[str] = None,
+        memory_mb: Optional[int] = None,
+        timeout_secs: Optional[int] = None,
+    ) -> Dict:
+        """Update the task with specified fields.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-object/update-task
+
+        Args:
+            name (string, optional): Name of the task
+            build (str, optional): Specifies the actor build to run. It can be either a build tag or build number.
+                                   By default, the run uses the build specified in the task settings (typically latest).
+            memory_mb (int, optional): Memory limit for the run, in megabytes. By default, the run uses a memory limit specified in the task settings.
+            timeout_secs: (int, optional): Optional timeout for the run, in seconds. By default, the run uses timeout specified in the task settings.
+            task_input (dict, optional): Task input dictionary
+
+        Returns:
+            The updated task
+        """
+        new_fields = {
+            "name": name,
+            "options": {
+                "build": build,
+                "memoryMbytes": memory_mb,
+                "timeoutSecs": timeout_secs,
+            },
+            "input": task_input,
+        }
+
+        return self._update(_filter_out_none_values_recursively(new_fields))
+
+    def delete(self) -> None:
+        """Delete the task.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-object/delete-task
+        """
+        return self._delete()
+
+    def start(
+        self,
+        *,
+        task_input: Optional[Dict[str, Any]] = None,
```

Comment:
I'm assuming everywhere that input is `Dict` only. @fnesveda is that correct? Thx